Mathlib/AlgebraicGeometry/Spec.lean

```diff
@@ -228,63 +267,84 @@ def Spec.locallyRingedSpaceMap {R S : CommRingCat} (f : R ⟶ S) :
-    rw [Spec.LocallyRingedSpace_map_val, Spec.SheafedSpace_map_comp]; rfl
+    rw [Spec.locallyRingedSpaceMap_val, Spec.sheafedSpaceMap_comp]; rfl
+set_option linter.uppercaseLean3 false in
 #align algebraic_geometry.Spec.LocallyRingedSpace_map_comp AlgebraicGeometry.Spec.locallyRingedSpaceMap_comp
 
 /-- Spec, as a contravariant functor from commutative rings to locally ringed spaces.
 -/
 @[simps]
 def Spec.toLocallyRingedSpace : CommRingCatᵒᵖ ⥤ LocallyRingedSpace where
   obj R := Spec.locallyRingedSpaceObj (unop R)
-  map R S f := Spec.locallyRingedSpaceMap f.unop
-  map_id' R := by rw [unop_id, Spec.LocallyRingedSpace_map_id]
-  map_comp' R S T f g := by rw [unop_comp, Spec.LocallyRingedSpace_map_comp]
+  map {R S} f := Spec.locallyRingedSpaceMap f.unop
+  map_id R := by dsimp; rw [Spec.locallyRingedSpaceMap_id]
+  map_comp {R S T} f g := by dsimp; rw [Spec.locallyRingedSpaceMap_comp]
+set_option linter.uppercaseLean3 false in
 #align algebraic_geometry.Spec.to_LocallyRingedSpace AlgebraicGeometry.Spec.toLocallyRingedSpace
 
 section SpecΓ
 
 open AlgebraicGeometry.LocallyRingedSpace
 
 /-- The counit morphism `R ⟶ Γ(Spec R)` given by `algebraic_geometry.structure_sheaf.to_open`.  -/
-@[simps (config := { rhsMd := Tactic.Transparency.semireducible })]
+@[simps!]
 def toSpecΓ (R : CommRingCat) : R ⟶ Γ.obj (op (Spec.toLocallyRingedSpace.obj (op R))) :=
   StructureSheaf.toOpen R ⊤
+set_option linter.uppercaseLean3 false in
 #align algebraic_geometry.to_Spec_Γ AlgebraicGeometry.toSpecΓ
 
-instance isIso_toSpecΓ (R : CommRingCat) : IsIso (toSpecΓ R) := by cases R;
-  apply structure_sheaf.is_iso_to_global
+instance isIso_toSpecΓ (R : CommRingCat) : IsIso (toSpecΓ R) := by
+  cases R; apply StructureSheaf.isIso_to_global
+set_option linter.uppercaseLean3 false in
 #align algebraic_geometry.is_iso_to_Spec_Γ AlgebraicGeometry.isIso_toSpecΓ
 
 @[reassoc]
 theorem Spec_Γ_naturality {R S : CommRingCat} (f : R ⟶ S) :
-    f ≫ toSpecΓ S = toSpecΓ R ≫ Γ.map (Spec.toLocallyRingedSpace.map f.op).op := by ext; symm;
+    f ≫ toSpecΓ S = toSpecΓ R ≫ Γ.map (Spec.toLocallyRingedSpace.map f.op).op := by
+  -- Porting note : `ext` failed to pickup one the three lemmas
+  refine RingHom.ext fun x => Subtype.ext <| funext fun x' => ?_; symm;
   apply Localization.localRingHom_to_map
+set_option linter.uppercaseLean3 false in
 #align algebraic_geometry.Spec_Γ_naturality AlgebraicGeometry.Spec_Γ_naturality
 
 /-- The counit (`Spec_Γ_identity.inv.op`) of the adjunction `Γ ⊣ Spec` is an isomorphism. -/
-@[simps hom_app inv_app]
+@[simps! hom_app inv_app]
 def specΓIdentity : Spec.toLocallyRingedSpace.rightOp ⋙ Γ ≅ 𝟭 _ :=
-  Iso.symm <| NatIso.ofComponents (fun R => asIso (toSpecΓ R) : _) fun _ _ => Spec_Γ_naturality
+  Iso.symm <| NatIso.ofComponents (fun R =>
+    -- Porting note : In Lean3, this `IsIso` is synthesized automatically
+    letI : IsIso (toSpecΓ R) := StructureSheaf.isIso_to_global _
+    asIso (toSpecΓ R)) fun {X Y} f => by
```

Comment:
```suggestion
    -- Porting note: In mathlib 3, the tactic block wasn't needed: `Spec_Γ_naturality` type-checked as term
    asIso (toSpecΓ R)) fun {X Y} f => by
```

Reply:
This behaviour is slightly confusing, `by exact Spec_Γ_naturality (R := X) (S := Y) f` works but `Spec_Γ_naturality (R := X) (S := Y) f` does not work